app/code/Magento/Theme/Block/Html/Topmenu.php

```diff
@@ -394,4 +394,17 @@ public function getMenu()
         }
         return $this->_menu;
     }
+    
+    /**
+     * Get cache key informative items
+     *
+     * @return array
+     * @since 100.1.0
+     */
+    public function getCacheKeyInfo()
+    {
+        $keyInfo = parent::getCacheKeyInfo();
+        $keyInfo[] = $this->getUrl('*/*/*', ['_current' => true, '_query' => '']);
```

Comment:
What do you think about using the current category in the cache key instead of the current URL?

Reply:
@slitvyachenko: do you mean the entire category object, or just some attribute of it? Which one then? We can't use the name, because the name isn't unique. Maybe the path and the store id?

Reply:
primary key